### PR TITLE
Make ldflags compatible with go modules with versions >= 2

### PR DESCRIPTION
**Pull request by fiunchinho**
## Checklist

- [x] Make yourself familiar with following readme sections:
    - [x] [Coding Standards](https://github.com/giantswarm/architect-orb#coding-guidelines).
    - [x] [Development](https://github.com/giantswarm/architect-orb#development).
    - [x] [Design and Goals](https://github.com/giantswarm/architect-orb#design-and-goals).
- [x] :warning: Update changelog in [CHANGELOG.md](https://github.com/giantswarm/architect-orb/tree/master/CHANGELOG.md).
- [ ] :warning: After the release update architect orb version in [.circleci/config.yml](https://github.com/giantswarm/architect-orb/tree/master/.circleci/config.yml).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
```
Comment (kopiczko) [inline CHANGELOG.md:10]:
```suggestion
### Fixed
```

**Comment by fiunchinho** [comment]
Done

File: CHANGELOG.md (hunk 1 above)
Comment (fiunchinho) [reply CHANGELOG.md:10]: I wrote the `|| true` part as `Changed`

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+- `go-build` command will fail if it fails to compile.
```
Comment (kopiczko) [inline CHANGELOG.md:12]: Commands are not important from the end user perspecitve.

```suggestion
- `go-build` job will fail if it fails to compile.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+### Fixed
+
+- The `go-test` command takes into consideration go modules when generating flags to inject values into the binary.
```
Comment (kopiczko) [inline CHANGELOG.md:16]:
```suggestion
- The `go-test` job takes into consideration go modules when generating flags to inject values into the binary.
```

**Pull request merged**